src/components/AddressSearch.js

```diff
@@ -109,12 +106,15 @@ const AddressSearch = (props) => {
                 label: props.label,
                 containerStyles: props.containerStyles,
                 errorText: props.errorText,
+                value: props.value,
```

Comment:
Using `value: props.value` plus doing `props.onChange({addressStreet: text});` during `onChangeText` keeps the value stored in the form's state synchronized with the input's value. We need to do this because before we were only updating the value on the form when the user clicked a result from the autocomplete. 

This change results in:
- Not needing the ref anymore
- The library clearing the input at the beginning becomes a problem. This is why I had to introduce the work around `skippedFirstOnChangeText`  again.

Reply:
opened an issue - let's see if anyone responds 🙃

Reply:
>This seems like a bug, no?
A bug or a poor design

>opened an issue - let's see if anyone responds 🙃

Thanks!